target/linux/ar71xx/files/arch/mips/ath79/mach-ligodlb-5.c

```diff
@@ -0,0 +1,118 @@
+/*
+ *  LigoWawe(Deliberant) LigoDLB Propeller 5 support
+ *
+ *  Copyright (C) 2010-2012 Gabor Juhos <juhosg@openwrt.org>
+ *  Copyright (C) 2011-2012 Anan Huang <axishero@foxmail.com>
+ *  Copyright (C) 2017 Robert Marko <robimarko@otvorenamreza.org>
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 2 as published
+ *  by the Free Software Foundation.
+ */
+ 
+#include <linux/pci.h>
+#include <linux/phy.h>
+#include <linux/platform_device.h>
+#include <linux/ath9k_platform.h>
+#include <linux/ar8216_platform.h>
+
+#include <asm/mach-ath79/ar71xx_regs.h>
+
+#include "common.h"
+#include "dev-ap9x-pci.h"
+#include "dev-eth.h"
+#include "dev-gpio-buttons.h"
+#include "dev-leds-gpio.h"
+#include "dev-m25p80.h"
+#include "dev-spi.h"
+#include "dev-wmac.h"
+#include "machtypes.h"
+
+#define LIGODLB_GPIO_BTN_RESET		17
+#define LIGODLB_GPIO_LED_LAN0		1
+#define LIGODLB_GPIO_LED_L1		2
+#define LIGODLB_GPIO_LED_L2		3
+#define LIGODLB_GPIO_LED_L3		4
+#define LIGODLB_GPIO_LED_L4		13
+
+#define LIGODLB_GPIO_LED_PWR		0
+
+#define LIGODLB_KEYS_POLL_INTERVAL	20	/* msecs */
+#define LIGODLB_KEYS_DEBOUNCE_INTERVAL	(3 * LIGODLB_KEYS_POLL_INTERVAL)
+
+static struct gpio_led ligdolb_leds_gpio[] __initdata = {
+	{
+		.name		= "ligodlb:green:power",
+		.gpio		= LIGODLB_GPIO_LED_PWR,
+		.default_state  = LEDS_GPIO_DEFSTATE_ON,
+		.active_low	= 1,
+	}, {
+		.name		= "ligodlb:green:lan0",
+		.gpio		= LIGODLB_GPIO_LED_LAN0,
+		.active_low	= 1,
+	}, {
+		.name		= "ligodlb:yellow:link1",
+		.gpio		= LIGODLB_GPIO_LED_L1,
+		.active_low	= 1,
+	}, {
+		.name		= "ligodlb:yellow:link2",
+		.gpio		= LIGODLB_GPIO_LED_L2,
+		.active_low	= 1,
+	}, {
+		.name		= "ligodlb:yellow:link3",
+		.gpio		= LIGODLB_GPIO_LED_L3,
+		.active_low	= 1,
+	}, {
+		.name		= "ligodlb:yellow:link4",
+		.gpio		= LIGODLB_GPIO_LED_L4,
+		.active_low	= 1,
+	},
+};
+
+static struct gpio_keys_button ligdolb_gpio_keys[] __initdata = {
+	{
+		.desc		= "reset",
+		.type		= EV_KEY,
+		.code		= KEY_RESTART,
+		.debounce_interval = LIGODLB_KEYS_DEBOUNCE_INTERVAL,
+		.gpio		= LIGODLB_GPIO_BTN_RESET,
+		.active_low	= 1,
+	}
+};
+
+static void __init ligdolb_5_setup(void)
+{
+	u8 *wifi_mac = (u8 *) KSEG1ADDR(0x1fff1002);
```

Comment:
This is not required, that's the regular MAC offset, you can call `ath79_register_wmac(ee, NULL);`.